spec/integration/aws_elasticsearch_domain_spec.rb

```diff
@@ -0,0 +1,119 @@
+require 'spec_helper'
+
+def policy(name)
+  <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Deny",
+      "Sid": "test-policy",
+      "Principal": {
+        "AWS": "*"
+      },
+      "Action": "es:*",
+      "Resource": "arn:aws:es:us-east-1::domain/#{name}/*"
+    }
+  ]
+}
+EOF
+end
+
+def all_options_domain(name)
+  aws_elasticsearch_domain name do
+    instance_type "m3.medium.elasticsearch"
+    instance_count 2
+    dedicated_master_enabled true
+    dedicated_master_type "m3.medium.elasticsearch"
+    dedicated_master_count 2
+    zone_awareness_enabled true
+    ebs_enabled true
+    volume_type "io1"
+    volume_size 35
+    iops 1000
+    automated_snapshot_start_hour 2
+    access_policies policy(name)
+    aws_tags key1: "value"
+  end
+end
+
+describe Chef::Resource::AwsElasticsearchDomain do
+  extend AWSSupport
+
+  let(:all_options_result) do
+    {created: true,
+     elasticsearch_cluster_config: {
+       instance_type: "m3.medium.elasticsearch",
```

Comment:
Is this the cheapest one available? (Or is the default the cheapest, and this is the second cheapest so we can test that we can change it?)